bucket/foxit-reader.json

```diff
@@ -18,10 +18,10 @@
     "innosetup": true,
     "checkver": {
         "url": "https://www.foxitsoftware.com/downloads/downloadForm.php?retJson=1&product=Foxit-Reader&platform=Windows&language=English",
-        "regex": "([\\d.]+).*(?<md5hash>([a-fA-F0-9]{32}))"
+        "regex": "([\\d.]+).*\\\\/(?<dlver>[\\d\\.]+)\\\\/(?<md5hash>([a-fA-F0-9]{32}))\\\\/(?<dlname>.+\\.exe)"
```

Comment:
Is there a reason not to use `jsonpath` along with `regex` to access only the data we need? It might save time compared to scanning the entire output.

Reply:
I'm not sure how to *use one `jsonpath` expression to extract multiple diffenent fields*, and the fields we need are from different paths (*say `$.package_info.version[0]` and `$.package_info.down`*). Even if it could be achieved that way, `regex` is still needed (*to give them different names so that we could interpolate them by their names*), and correct me if I am wrong, by using `regex` alone it will only consume (*process*) the entire json string once yet using `jsonpath` together with `regex` might resulting in another round of consuming the  extracted fields after consuming the entire json string (*to deserialize it*) while still having to take an effort to compile the `regex` either.

Reply:
True enough on the multiple JSONPaths. It's a shame we can't specify a JSONPath for each piece of data we need to extract - it would certainly make sense given that JSONPath is even an option. It would also make extraction less reliant on a consistently ordered JSON output, as well as improving the readability of any resulting regexes

As for the comment about regexes versus JSONPath plus regex: Regexes are a wonderful tool for matching and extracting/manipulating strings, however it's definitely one of the slower methods of string matching: Especially on longer strings with less well optimised expressions. I would be very surprised if regex scanning an entire mess of JSON was faster than pulling out individual values - but hey, maybe it's worth comparing, performance-wise.

(Although this debate does remind me of [this famous StackOverflow answer about parsing HTML with regexes...](https://stackoverflow.com/a/1732454) :) )